core/generator.js

```diff
@@ -104,7 +105,11 @@ Blockly.Generator.prototype.workspaceToCode = function(workspace) {
       // Top-level blocks don't care about operator order.
       line = line[0];
     }
-    if (line) {
+
+    if(line == undefined){
```

Comment:
What if the unsupported block was not a top block, whether a next block or a value block? It doesn't look like your code catches these cases.

Reply:
I could not understand what is implied from a 'top block'.

Reply:
And how can I make a distinction in the code?